code/game/objects/items/weapons/traps.dm

```diff
@@ -607,55 +523,66 @@
 						/mob/living/simple_animal/hostile/carp, /mob/living/simple_animal/hostile/bear, /mob/living/simple_animal/hostile/alien, /mob/living/simple_animal/hostile/giant_spider,
 						/mob/living/simple_animal/hostile/commanded/dog, /mob/living/simple_animal/hostile/retaliate/cavern_dweller, /mob/living/carbon/human/)
 
-/obj/item/weapon/trap/animal/large/attack_hand(mob/user as mob)
-	return
+/obj/item/weapon/trap/animal/large/attack_hand(mob/user)
+	if(user.loc == loc)
+		to_chat(user, span("warning", "You can't do that from the inside!"))
```

Comment:
Can't you go onto the trap's tile without triggering it? If so, this isn't a good check.